2371/2/application.rb

```diff
@@ -0,0 +1,87 @@
+require 'optparse'
+require File.expand_path(File.dirname(__FILE__) + '/rap_battles_manager')
+require File.expand_path(File.dirname(__FILE__) + '/constants')
+
+# The Application responsible for start searching info
+class Application
+  attr_reader :options
+  def initialize
+    @thread = Thread.new do
+      0.step(1000, 5) do |step|
+        printf("\rSearching:  %-20s", '*' * (step / 5))
+        sleep(0.5)
+      end
+    end
+    @manager = RapBattlesManager.new
+    @options = {}
+  end
+
+  def show_authors_bad_w_rating(number)
```

Comment:
It took some time to figure out that `w` is `word`. Please, be explicit in naming.